mec172xevb_assy6906: Add ADC test support for the Microchip board

hunterhu3000 opened the pull request:
Without this config test/drivers/adc/adc_api will fail
on mec172xevb_assy6906. The build and run commands are:
west build -p always -b mec172xevb_assy6906 tests/drivers/adc/adc_api/
west flash

Signed-off-by: Hu Zhenyu <zhenyu.hu@intel.com>

anangl commented:
This commit does not actually change anything in dts, so it should not be titled "dts: mec172xevb_assy6906: Add support for the Microchip board". Something like "tests: drivers: adc_api: Add configuration for mec172xevb_assy6906" seems to be more suitable.

MaureenHelm reviewed (changes requested):
Need to add adc to the supported list in `boards/arm/mec172xevb_assy6906/mec172xevb_assy6906.yaml`. Without it, twister doesn't execute the adc_api test for this board.

hunterhu3000 committed:
tests: drivers: adc_api: Add configuration for mec172xevb_assy6906

Enable the tests/drivers/adc/adc_api/ test for mec172xevb_assy6906
on both west and twister

Signed-off-by: Hu Zhenyu <zhenyu.hu@intel.com>
